src/engine/Union.cpp

```diff
@@ -292,3 +334,185 @@ std::unique_ptr<Operation> Union::cloneImpl() const {
   }
   return copy;
 }
+
+// _____________________________________________________________________________
+std::shared_ptr<Operation> Union::createSortedVariant(
+    const vector<ColumnIndex>& sortOrder) const {
+  return std::make_shared<Union>(_executionContext, _subtrees.at(0),
+                                 _subtrees.at(1), sortOrder);
+}
+
+namespace {
+struct Wrapper {
+  const IdTable& idTable_;
+  const LocalVocab& localVocab_;
+};
+Result::IdTableVocabPair& moveOrCopy(Result::IdTableVocabPair& element) {
+  return element;
+}
+Result::IdTableVocabPair moveOrCopy(const Wrapper& element) {
+  return {element.idTable_.clone(), element.localVocab_.clone()};
+}
+}  // namespace
+
+// _____________________________________________________________________________
+// Always inline makes makes a huge difference on large datasets.
+template <size_t SPAN_SIZE>
+AD_ALWAYS_INLINE bool Union::isSmaller(const auto& row1,
+                                       const auto& row2) const {
+  using StaticRange = std::span<const ColumnIndex, SPAN_SIZE>;
```

Comment:
How do you handle the case, that we have more than 10 sort columns, and callFixedSize returns 0?
Oh... I see, you probably use span with dynamic_extent, my bad.